extensions/db/src/main/java/org/mule/extension/db/internal/operation/BaseDbOperations.java

```diff
@@ -97,6 +102,22 @@ protected void validateQueryType(QueryTemplate queryTemplate, List<QueryType> va
     }
   }
 
+
+  protected void validateNoParameterTypeIsUnused(Query query, List<ParameterType> parameterTypes) {
+    Set<String> params = query.getQueryTemplate().getParams().stream().map(QueryParam::getName).collect(Collectors.toSet());
+    Set<ParameterType> unusedTypes =
+        parameterTypes.stream().filter(type -> !params.contains(type.getKey())).collect(Collectors.toSet());
+
+    if (!unusedTypes.isEmpty()) {
+      throw new IllegalArgumentException(format("Query parameters [%s] were unused during query resolution, please remove them",
```

Comment:
this error message is not descriptive of the actual problem. Please rephrase to something like `query defines parameters `%s` --no brackets needed- but they're not present in the query. `